### DISCUSSION TITLE
Entities for controlling mode of operation

### DISCUSSION THREAD

**Discussion by cypher0815** [mbuchber/ha_heliotherm#35] [Ideas]
The only entities for controlling the heat pump (at least Brötje Neo 18 ) are:

climate.brotje_neo_raum_solltemperatur,
select.brotje_neo_betriebsart,
select.brotje_neo_mkr_1_betriebsart and
select.brotje_neo_mkr_2_betriebsart

The approach when using NEO-RKM is completely different through creating heating demands controlling the return flow temperature or compressor speed or power consumption which controls compressor speed indirectly. This is described in detail in the NEO-RKM manual.

It would be great if these controlling parameters could be added.

**Comment by mbuchber** [comment]
You would need to elaborate to me more clearly and send me the docs - I only have one heatpump in my home, so I cannot verify all that stuff,
Best regards,
 Michael

**Comment by cypher0815** [comment]
Hi Michael,

"heating demand profiles" are described in detail in the [[Manual_neo-rkm.pdf]](https://github.com/mbuchber/ha_heliotherm/blob/main/docs/Manual_neo-rkm.pdf) in chapter 6.6.5.
There are only a few registers that need to be written. The only things that are missing are actions for heating demands in the integration.
I tried to add a climate entity for rl_soll and it works manually but climate entities don´t work in automations. As a beginner i am not able to add services to integrations in HA.

Her are some examples:
Request for hot water:
  async def set_ww_bereitung(self, temp_min: float, temp_max: float):
      if temp_min is None or temp_max is None:
          return
      temp_max_int = int(temp_max * 10)
      temp_min_int = int(temp_min * 10)
      self._client.write_register(address=105, value=temp_max_int, slave=1)
      self._client.write_register(address=106, value=temp_min_int, slave=1)
      await self.async_refresh_modbus_data()

Request for return target temperature:
    async def set_rl_soll(self, temperature: float):
        if temperature is None:
            return
        temp_int = int(temperature * 10)
        temp_activate_rl_soll = 1
        self._client.write_register(address=102, value=temp_int, slave=1)
        self._client.write_register(address=103, value=temp_activate_rl_soll, slave=1)
        await self.async_refresh_modbus_data()
If you have any questions or need a tester feel free to contact me.

**Reply by kobius77** [reply]
i'm using node-red for testing such things, there it's relatively easy to implement and communicate over modbus without (or little) code.

but i still couldn't get these manual/pv demands working on my machine - i get it to show "pv demand" on the display, but whatever i try it doesn't start ;(

**Comment by cypher0815** [comment]
According to the documentation you need to enable it:
  Anforderungsprofil 2: Anforderung zur VerdichterDrehzahlvorgabe
(Regulierung der Heizleistung) Um diese Variante der externen Anforderung
an die Wärmepumpe zu realisieren, sind weitere Einstellungen an der
Wärmepumpenregelung NEO-RWP und am NEO-RKM vorzunehmen: NEO-RKM:
Einstellungen → Grundbedienung → PV (Photovoltaik) → Auswahl auf OFF
NEO-RWP: Hauptmenü → “WNA” → FU extern auf ON....

Am Mo., 13. Jan. 2025 um 11:21 Uhr schrieb drahdiwaberl <
***@***.***>:

**Reply by kobius77** [reply]
ja, alles gemacht. dein brötje-manual ist etwas klarer, drum hab ichs letztens nochmal probiert. aber ausser der anzeige "PV Anforderung" und der Möglichkeit die RLsoll einzustellen tut sich einfach nix. Erst wenn ich wieder über den Modus (zb Party) eine Anforderung "Heizung" provoziere startet die Kiste.. :/

**Comment by cypher0815** [comment]
Hast Du mal die Änderung in meinem Fork
<https://github.com/cypher0815/ha_heliotherm> ausprobiert?
[image: image.png]
Meine WP läuft an, sobald ich die R-Solltemperatur erhöhe auf einen Wert
der höher als 5K vom Pufferspeicher ist.
Stell den Modus mal auf Dauerbetrieb und teste dann.

Am Mo., 13. Jan. 2025 um 12:20 Uhr schrieb drahdiwaberl <
***@***.***>:
